examples/research_projects/wav2vec2/run_wav2vec2_eval_with_lm.py

```diff
@@ -0,0 +1,292 @@
+
+            self.decoder = LexiconDecoder(
+                self.decoder_opts,
+                self.trie,
+                self.lm,
+                self.silence,
+                self.blank,
+                self.unk_word,
+                self.asg_transitions,
+                args.unit_lm,
+            )
+        else:
+            assert args.unit_lm, "lexicon free decoding can only be done with a unit language model"
+            from flashlight.lib.text.decoder import LexiconFreeDecoder, LexiconFreeDecoderOptions
+
+            d = {w: [[w]] for w in tgt_dict}
+            self.word_dict = create_word_dict(d)
+            self.lm = KenLM(args.kenlm_model, self.word_dict)
+            self.decoder_opts = LexiconFreeDecoderOptions(
+                beam_size=args.beam,
+                beam_size_token=int(len(tgt_dict)),
+                beam_threshold=args.beam_threshold,
+                lm_weight=args.lm_weight,
+                sil_score=args.sil_weight,
+                log_add=False,
+                criterion_type=self.criterion_type,
+            )
+            self.decoder = LexiconFreeDecoder(self.decoder_opts, self.lm, self.silence, self.blank, [])
+
+    def decode(self, emissions):
+        B, T, N = emissions.size()
+        hypos = []
+        for b in range(B):
+            emissions_ptr = emissions.data_ptr() + 4 * b * emissions.stride(0)
+            print("in decode")
+            print(self.decoder)
+            results = self.decoder.decode(emissions_ptr, T, N)
+            print("after decode")
+            nbest_results = results[: self.nbest]
+            hypos.append(
+                [
+                    {
+                        "prefix": self.get_prefix(result.tokens),
+                        "score": result.score,
+                        "words": [self.word_dict.get_entry(x) for x in result.words if x >= 0],
+                    }
+                    for result in nbest_results
+                ]
+            )
+        return hypos[0]
+
+
+def main():
+    parser = HfArgumentParser((EvaluationArguments))
+
+    eval_args = parser.parse_args_into_dataclasses()
+    eval_args = eval_args[0]
+    configure_logger(eval_args)
+
+    processor = Wav2Vec2Processor.from_pretrained(eval_args.model_name_or_path)
+    model = Wav2Vec2ForCTC.from_pretrained(eval_args.model_name_or_path)
+
+    def map_to_result(batch):
+        if eval_args.use_cuda:
+            model.to("cuda")
+            input_values = processor(
+                batch["speech"], sampling_rate=batch["sampling_rate"], return_tensors="pt"
+            ).input_values.to("cuda")
+        else:
+            input_values = processor(
+                batch["speech"], sampling_rate=batch["sampling_rate"], return_tensors="pt"
+            ).input_values
+
+        with torch.no_grad():
+            logits = model(input_values).logits
+
+        target_dictionary = [t for t in processor.tokenizer.get_vocab().keys()]
```

Comment:
Testing W2lViterbiDecoder i figure out that this list must by ordered by original token index.
```suggestion
        vocab = processor.tokenizer.get_vocab()
        target_dictionary = sorted(vocab, key=lambda k: vocab[k])
```